quickwit-index-config/resources/tests/index_object/hdfs-logs.json

```diff
@@ -0,0 +1,35 @@
+{
+    "index_id": "hdfs-logs",
+    "index_uri": "s3://quickwit-indexes/hdfs-logs",
+    "indexing_settings": {
+        "commit_timeout_secs": 60,
+        "split_max_num_docs": 10000000,
+        "merge_policy": {
+            "demux_factor": 6,
+            "merge_factor": 10,
+            "max_merge_factor": 12,
+            "min_level_num_docs": 100000
+        },
+        "resources": {
+            "num_threads": 1,
+            "heap_size": "1G"
+        }
+    },
+    "sources": [
```

Comment:
What's today in indexconfig (schema stuff most notably) will end up nest in this object too or not?

Reply:
In the current spec in Notion, `index_mapping` is at the root of the object. I see what you're getting at though. Depending on the source, docs could be different, but we would like to map them to the same index schema eventually.

We could define an index schema at the root and, in each source, a set of mappings/transformations (pseudo-DSL): 

```yaml
[...]
index_schema:
  - ts:
    type: u64
  - title:
    type: text
  - body:
    type: text

sources:
  - source_id: hdfs-logs-kafka-source
    source_type: kafka
    transforms:
      - rename:
        source: content
        target: body
      - drop:
        source: url
    params:
      bootstrap.servers: host:9092
      topic: cloudera-cluster-logs
    
  - source_id: hdfs-logs-kinesis-source
    source_type: kinesis
    transforms:
      - rename:
        source: timestamp
        target: ts
      - drop:
        source: url
    params:
      stream_name: emr-cluster-logs
```

We'll have to put more thought into it.